### PR TITLE
clarify "configure" messages from Cabal

### PR DESCRIPTION
**Pull request by geekosaur**
**Template Α: This PR modifies `cabal` behaviour**

Include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#other-conventions).
* [x] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [x] The documentation has been updated, if necessary.
* [ ] [Manual QA notes](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#qa-notes) have been included.
* [ ] Tests have been added. (*Ask for help if you don’t know how to write them! Ask for an exemption if tests are too complex for too little coverage!*)

### CODE REVIEW

**Comment by geekosaur** [comment]
I used template A because error messages are user-facing. I'm not sure how to test these though, as the only way I know to trigger them is a possible cabal-install bug suffered by @Kleidukos around a week ago.

**Comment by Kleidukos** [comment]
I have this bug when using cabal-head but not a released version. Thanks for handling this @geekosaur

**Comment by geekosaur** [comment]
Sadly I haven't fixed the actual bug, just made the error message a little less confusing. I assume there's a mismatch between what `Distribution/Simple/Configure.hs` serializes to `setup-config` and what it expects when it reads it back in later.

**Comment by geekosaur** [comment]
BTW @Kleidukos can you file a bug about your error so we have a record of it? I may or may not be able to get to it this weekend and I'll be gone on Monday. AFAICT you only mentioned it on Matrix.

**Comment by andreabedini** [comment]
LGTM, is `Setup` clear enough in context?

**Comment by geekosaur** [comment]
I still don't see a bug corresponding to this, and have no idea whether the original problem still exists, nor how I would go about testing this.

**Comment by ulysses4ever** [comment]
@geekosaur do you have a plan for this? SHould it be turned into non-draft?

**Comment by Kleidukos** [comment]
I completely missed this. I'll see if it's still a thing.

**Comment by geekosaur** [comment]
As far as I'm concerned it's ready to go. But all it does is change some messaging, since Cabal confusingly tells you that you need to reconfigure (which is strictly correct, but if you're using cabal-install then "reconfigure" means something else entirely).

File: changelog.d/configure-messages [hunk 1]
```diff
@@ -0,0 +1,13 @@
+synopsis: clarify Cabal "configure messages
+packages: Cabal
+prs: #9476
+
+synopsis: {
+
+	Cabal can issue a number of error messages referencing "Setup configure",
+	but it simply references "configure" which could mean any of three
+	things (Setup configure, the package's "configure" script, or "cabal
+	configure". This has recently caught out even Cabal devs. Clarify these
```
Comment (jasagredo) [inline changelog.d/configure-messages:10]:
```suggestion
	configure"). This has recently caught out even Cabal devs. Clarify these
```

File: changelog.d/configure-messages [hunk 2]
```diff
@@ -0,0 +1,13 @@
+synopsis: clarify Cabal "configure messages
```
Comment (jasagredo) [inline changelog.d/configure-messages:1]:
```suggestion
synopsis: clarify Cabal "configure" messages
```

**Review by jasagredo — APPROVED** [review]

**Review by Kleidukos — APPROVED** [review]

**Pull request merged**